waspc/data/Generator/templates/universal/types.ts

```diff
@@ -0,0 +1,16 @@
+// This is a helper type used exclusively for DX purposes. It's a No-op for the
+// compiler, but expands the type's representatoin in IDEs (i.e., inlines all
+// type constructors) to make it more readable for the user.
+//
+// It expands this SO answer to funcitons: https://stackoverflow.com/a/57683652
+export type Expand<T> = T extends (...args: infer A) => infer R
+  ? (...args: A) => R
+  : T extends infer O
+  ? { [K in keyof O]: O[K] }
+  : never
+
+// TypeScript's native Awaited type exhibits strange behavior in VS Code (see
+// todo for details). Until it's fixed, we're using our own type for this.
+export type _Awaited<T extends Promise<unknown>> = T extends Promise<infer V>
+  ? V
+  : never
```

Comment:
Take a look at [this playground](https://www.typescriptlang.org/play?noImplicitAny=false&strictNullChecks=false&strictFunctionTypes=false&strictPropertyInitialization=false&strictBindCallApply=false&noImplicitThis=false&noImplicitReturns=false&alwaysStrict=false&ts=4.9.5#code/C4TwDgpgBAYlC8UAUBDATgcwM4C4oDsBXAWwCMI0AaKAYwHt9gIAPYPIsigSigQD4oABTR1iASywQAPIXwBrfHQDu+PgChQkKAHEEydNnYQAbhWr1GLNgRLk0PeAOGiJ02QuWq1G8NABKEMCEaPhwiACCSihiTAAmUgFBIQAqvlIwfAIA9FlQWAAWdIQANrFQ5FAA5O6KKpXUisBVKPgglT5aicH4uhFRMRDxXSlp2plQOXmFJWUV1fK1+PUEdE2VLW1AA) and try the same snippet in VS Code. This strange behavior causes a problem for operations with unspecified inputs.

#### Example
Let's say your backend query definition looks like this:
```typescript
export const getTasks: GetTasks = async (_args, context) => {
  // ...
}
```
This is what we want TypeScript to infer on the client:

<img src="https://user-images.githubusercontent.com/22752307/230365910-ec195ef4-fbc2-45f1-9440-0abbc6736729.png" width="500px" />

Unfortunately, if we use the built-in `Awaited` type, this happens:

<img src="https://user-images.githubusercontent.com/22752307/230365715-382257cd-a0f5-449b-a2fb-6d9e8d0409c0.png" width="500px" />